### PR TITLE
work around ResultLog support for pytest>=6.1

### LINKED ISSUE #128: _pytest.resultlog was removed in pytest 6.1.0
`pytest-rerunfailures` imports `_pytest.resultlog` but the module was removed in https://github.com/pytest-dev/pytest/commit/ef946d557cc7b7f030805b94c1dbae51f39fcce4 .

As a result, it is no longer possible to use `pytest-rerunfailures` with pytest 6.1.0.

The suggested replacement is [pytest-reportlog](https://github.com/pytest-dev/pytest-reportlog).

### PR DESCRIPTION
**Pull request by ntessore**
This is a workaround that closes #128. It does not address potential integration with the reportlog plugin.

### CODE REVIEW

**Review by gnikonorov — APPROVED** [review]
As an immediate fix to stop the fatal error, I think this is OK. I will defer to @sallner and/or @icemac for the merge.

@ntessore if this gets merged can you open a separate issue to track integration with `reportlog` so it doesn't get lost?

**Comment by gnikonorov** [comment]
Could you also add a changelog entry to this PR, @ntessore? It can be added in the  [CHANGES.rst](https://github.com/pytest-dev/pytest-rerunfailures/blob/master/CHANGES.rst) file

**Comment by ntessore** [comment]
@gnikonorov I'm really not qualified to open an issue on pytest plugin interoperability, I wouldn't even know if that should be an issue here or in pytest-dev/pytest-reportlog.

File: CHANGES.rst [hunk 1]
```diff
@@ -4,7 +4,10 @@ Changelog
 9.2 (unreleased)
 ----------------
 
-- Nothing changed yet.
+Other changes
++++++++++++++
+
+- Deprecated `--result-log` was removed in pytest 6.1.0.
```
Comment (gnikonorov) [inline CHANGES.rst:10]: Nit: can we change the wording a bit to something like 'Deprecate `--result-log`, as it was removed in  pytest 6.1.0 )'

**Review by gnikonorov — CHANGES_REQUESTED** [review]
Just a small nitpickish comment on the changelog text

**Comment by gnikonorov** [comment]
> @gnikonorov I'm really not qualified to open an issue on pytest plugin interoperability, I wouldn't even know if that should be an issue here or in pytest-dev/pytest-reportlog.

@ntessore I'll open one then. Thank you for fixing the crash

File: CHANGES.rst (hunk 1 above)
Comment (icemac) [inline CHANGES.rst:10]:
```suggestion
- Deprecate ``--result-log``, as it was removed in pytest 6.1.0
```

**Review by nicoddemus — APPROVED** [review]

**Comment by sallner** [comment]
I am unhappy with the fact, that we have now 3 ways of handling the resultlog in this plugin, but it seems that a bugfix release might be appropriate to make various CIs green again.

**Review by sallner — APPROVED** [review]
I will merge the changes and cut a release as 9.1.1. We can clean up the rest afterwards.

**Comment by icemac** [comment]
> I will merge the changes and cut a release as 9.1.1. We can clean up the rest afterwards.

Maybe it should become a 9.2 or even 10.0 as the `--result-log` command line option is no linger supported when this plug-in is used together with pytest >= 6.1.

**Comment by sallner** [comment]
@ntessore Thanks for the work, version 9.1.1 is out with the workaround. https://pypi.org/project/pytest-rerunfailures/9.1.1/ Let's see, what we will do with the clean up, and if we need to support resultlog in the future.

**Comment by hugovk** [comment]
Thank you all for the quick fix and release!

**Pull request merged**